diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba5d34e..c4e1809 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,10 +20,11 @@ if(NOT EXISTS ${UNICORN_HOME})
     set(UNICORN_HOME ${CMAKE_CURRENT_SOURCE_DIR}/unicorn)
 endif()
 
-if(CMAKE_BUILD_TYPE MATCHES DEBUG)
-    set (UNICORN_DEBUG yes)
+string(TOUPPER ${CMAKE_BUILD_TYPE} BUILD_TYPE_UPPER)
+if(BUILD_TYPE_UPPER STREQUAL "DEBUG")
+    set(UNICORN_DEBUG yes)
 else()
-    set (UNICORN_DEBUG no)
+    set(UNICORN_DEBUG no)
 endif()
 
 message(STATUS "Found CAPSTONE at " ${CAPSTONE_HOME})
