diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba5d34e..53320ce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,7 @@ include(ExternalProject)
 project(ocx-qemu-arm)
 
 set(CMAKE_CXX_STANDARD 11)
+option(OCX_QEMU_ARM_BUILD_TESTS OFF)
 
 set(CAPSTONE_HOME $ENV{CAPSTONE_HOME})
 if(NOT EXISTS ${CAPSTONE_HOME})
@@ -83,7 +84,9 @@ target_link_libraries(ocx-qemu-arm ${UNICORN_LIB} capstone-static)
 
 install(TARGETS ocx-qemu-arm DESTINATION lib)
 
-enable_testing()
-add_test(NAME ocx-qemu-arm
-         COMMAND $<TARGET_FILE:ocx-test-runner>
-                 $<TARGET_FILE:ocx-qemu-arm> Cortex-A53)
+if (OCX_QEMU_ARM_BUILD_TESTS)
+    enable_testing()
+    add_test(NAME ocx-qemu-arm
+            COMMAND $<TARGET_FILE:ocx-test-runner>
+                    $<TARGET_FILE:ocx-qemu-arm> Cortex-A53)
+endif()
